boards/shared/Test/Src/Test_SharedStateMachine.cpp

```diff
@@ -0,0 +1,115 @@
+#include <gtest/gtest.h>
+#include <fff.h>
+
+#include "Test_Utils.h"
+#include "App_TestWorld.h"
```

Comment:
It is possible to move this include inside `extern C` to better match other test binaries? Or would that break compilation?

Reply:
That will break compilation, this is basically the issue we discussed above.